src/main/scala/mesosphere/marathon/api/TaskKiller.scala

```diff
@@ -100,8 +103,15 @@ class TaskKiller @Inject() (
       toKill = appTasks
     )
 
-    appTasks.keys.find(id => !instanceTracker.hasSpecInstancesSync(id))
-      .map(id => Future.failed(PathNotFoundException(id)))
-      .getOrElse(killTasks)
+    val idHasSpec = Future.sequence(appTasks.keys.map(id => async {
```

Comment:
I tried to keep old logic but you suggestion makes code cleaner. Fixed it. Thanks